### PR TITLE
App initialization

### PR DESCRIPTION
**Pull request by FinnStutzenstein**
Used for internal apps as well as for plugins. The pluginpart is
currently missing, in fact that the main OpenSlides part is more
important. Apps can give their models and mainmenu entries.

Routes are not enabled, because the routes have to be static for webpack
to build the bundles. If we want to keep lazy loading, I see no
possibility to encapsulate the routes from the site-routing module.

### CODE REVIEW

**Comment by FinnStutzenstein** [comment]
@tsiegleauq Please review.

File: client/src/app/app.module.ts [hunk 1]
```diff
@@ -21,6 +22,11 @@ import { LoginModule } from './site/login/login.module';
 export function HttpLoaderFactory(http: HttpClient): PruningTranslationLoader {
     return new PruningTranslationLoader(http);
 }
+
+export function AppLoaderFactory(appLoadService: AppLoadService): () => Promise<void> {
```
Comment (tsiegleauq) [inline client/src/app/app.module.ts]: needs comment

File: client/src/app/core/services/app-load.service.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { CollectionStringModelMapperService } from './collectionStringModelMapper.service';
+import { MediafileAppConfig } from '../../site/mediafiles/mediafile.config';
+import { MotionsAppConfig } from '../../site/motions/motions.config';
+import { SettingsAppConfig } from '../../site/settings/settings.config';
+import { AgendaAppConfig } from '../../site/agenda/agenda.config';
+import { AssignmentsAppConfig } from '../../site/assignments/assignments.config';
+import { UsersAppConfig } from '../../site/users/users.config';
+import { MainMenuService } from './main-menu.service';
+
+const appConfigs: AppConfig[] = [
```
Comment (tsiegleauq) [inline client/src/app/core/services/app-load.service.ts]: comment:
what is `AppConfig[]` going to do and why is this necessary?

File: client/src/app/site/core/core.module.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { SharedModule } from '../../shared/shared.module';
+import { PrivacyPolicyComponent } from './components/privacy-policy/privacy-policy.component';
+import { StartComponent } from './components/start/start.component';
+import { LegalNoticeComponent } from './components/legal-notice/legal-notice.component';
+
+@NgModule({
+    imports: [CommonModule, MotionsRoutingModule, SharedModule],
+    declarations: [PrivacyPolicyComponent, StartComponent, LegalNoticeComponent]
+})
+export class CoreModule {}
```
Comment (tsiegleauq) [inline client/src/app/site/core/core.module.ts]: we have two `CoreModule`s now. As discussed earlier, avoiding repeating module names would be highly appreciated.
Easy alternatives would be "main" or "master".

we could also rename the 'site' folder to something else or include multiple feature-folders.

i.e.

- core
- shared
- access
  - start
  - login
  - privacy-policy
  - legal-notie
- projector
  - ( ... )
- main
  - motions
  - agenda
  - ( ... )

File: client/src/plugins.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export const plugins: string[] = [
+
+]
```
Comment (tsiegleauq) [inline client/src/plugins.ts]: missing empty line ;-)

File: client/src/app/site/core/core.module.ts (hunk 3 above)
Comment (jsaalfeld) [reply client/src/app/site/core/core.module.ts]: Isn't "master" a bit discriminating? (see https://bugs.python.org/issue34605 for further reference).

BTT:  I'm all for renaming the `core` in sites to something else, like `corn-hub`, `nitty-gritty` or `heart` or `common`, please do not just add a prefix, like `os-core`. The reason for keeping `core` in the main folder for me would be, that it may hurt us in a later upgrade process, since the standard-folders are then renamend.

**Comment by FinnStutzenstein** [comment]
@tsiegleauq Renamed core->common. Should be mergeable now.

File: client/src/app/core/services/app-load.service.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+@Injectable({
+    providedIn: 'root'
+})
+export class AppLoadService {
+    public constructor(
+        private modelMapper: CollectionStringModelMapperService,
+        private mainMenuService: MainMenuService
+    ) {}
+
+    public async loadApps() {
```
Comment (tsiegleauq) [inline client/src/app/core/services/app-load.service.ts]: linter tells me: "expected call signature"
